crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1645,17 +1638,14 @@ impl RpcServerConfig {
         self.build(&modules).await?.start(modules).await
     }
 
-    /// Convenience function to do middleware creation in one spot
-    pub fn create_middleware(
-        &self,
-        cors: Option<String>,
-        jwt_secret: Option<JwtSecret>,
-    ) -> MiddlewareServiceBuilder {
-        let cors_layer: Option<CorsLayer> =
-            cors.as_deref().map(cors::create_cors_layer).and_then(|res| res.ok()).or_else(|| None);
-        let jwt_auth: Option<AuthLayer<JwtAuthValidator>> =
-            jwt_secret.map(|secret| AuthLayer::new(JwtAuthValidator::new(secret)));
-        tower::ServiceBuilder::new().option_layer(cors_layer).option_layer(jwt_auth)
+    /// Creates the [CorsLayer] if any
```

Comment:
nice! this is awesome. i also didn't like the `MiddlewareServiceBuilder` type. next time i'll know i can avoid triggering clippy by creating smaller more atomic functions